src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -743,6 +777,16 @@ private Status TryScanMethod(MethodIL methodIL, Value[] parameters, Stack<Method
                                         return Status.Fail(methodIL.OwningMethod, opcode);
                                 }
                             }
+                            else if (popped.ValueKind == StackValueKind.ByRef
+                                && (opcode == ILOpcode.conv_i || opcode == ILOpcode.conv_u)
+                                && (reader.PeekILOpcode() is >= ILOpcode.ldind_i1 and <= ILOpcode.ldind_ref) || reader.PeekILOpcode() == ILOpcode.ldobj)
```

Comment:
```suggestion
                                && (reader.PeekILOpcode() is (>= ILOpcode.ldind_i1 and <= ILOpcode.ldind_ref) or ILOpcode.ldobj)
```
Pattern matching all the way.